### PR TITLE
Add namelists for c96 baroclinic setup for benchmarking [DSL-315]

### PR DESCRIPTION
**Pull request by oelbert**
Adding namelists for C96 runs

### CODE REVIEW

**Comment by rheacangeo** [comment]
Very cool! Sadly it failed weirdly to generate the serialization data (https://jenkins.ginko.ch/job/fv3gfs-fortran-generate-test-data/186/console), but hopefully we can figure it out.
15:17:53 cat: /rundir/env.out: No such file or directory
15:17:53 >>>>> Aborting
huh?

**Comment by ofuhrer** [comment]
Some comments from my side:
- Should we not default all I/O (including diagnostics, restarts, etc.) to off since the main focus of these namelists is to collect performance numbers which should not be polluted by I/O. Should it not be an active decision to switch on I/O for validation purposes and longer runs (we will only be doing 60 timesteps or so)?
- If `dycore_only=.true.` and `fv_sg_adj=0` we are all set an we need to run is `fv_dynamics`. Is the idea we start with that and potentially switch on `fv_sg_adj=X` once we have the code ported? Or is sub-grid scale mixing not doing anything within the first couple of timesteps of the baroclinic test case?
- Is the argument for `do_sat_adj=.false.` the fact that humidity is close to zero anyways and it would do nothing?
- What does `fill_dp=.true./.false.` do? Why has it flipped?
- Are you sure about `k_split=7; n_split=8`? Are those the values they use for DYAMOND (c3072)?
  image
- What is the plan for `dt_atmos` for c96, c128, c192, c256? I see you left it constant from c48 to c96. What is the timestep they are using for the DYAMOND (c3072) runs?
- Are the settings for the sponge representative of the DYAMOND runs (c3072)?

**Review by ofuhrer — CHANGES_REQUESTED** [review]
I made some comments with questions and suggestions.

File: tests/serialized_test_data_generation/configs/c96_54ranks_baroclinic.yml [hunk 1]
```diff
@@ -0,0 +1,327 @@
+data_table: default
```
Comment (rheacangeo) [inline tests/serialized_test_data_generation/configs/c96_54ranks_baroclinic.yml:1]: before this is merged in we should make sure PR 118 is merged to master or here and also generate the c96_54ranks_baroclinic data (to make sure that won't crash jenkins every time that runs). Depending on how long this one takes, we might want to change the frequency of the jenkins validation. But the c96_6ranks was pretty quick.

File: tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    - 0
+    days: 0
+    dt_atmos: 225
+    dt_ocean: 225
+    hours: 0
+    memuse_verbose: false
+    minutes: 3
+    months: 0
+    ncores_per_node: 32
+    seconds: 45
```
Comment (rheacangeo) [inline tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml:83]: How many timesteps do we want these configurations to specify? For the baseline run, these will have to change... but we probably don't want 60 timesteps or whatever here running in jenkins... I wonder how long that takes. Are there any other namelist parameters that would have to change for the performance baseline and comparison runs?

File: tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    d_ext: 0.0
+    dddmp: 0.5
+    delt_max: 0.002
+    dnats: 1
+    do_sat_adj: false
+    do_vort_damp: true
+    dwind_2d: false
+    external_ic: false
+    fill: true
+    fill_dp: false
```
Comment (rheacangeo) [inline tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml:124]: I am confused as to how this is true for the other configurations and doesn't cause validation issues, we didn't port mix_dp -- looks like it only does something when delp < dpmin, so maybe we have just gotten lucky? It seems like a setting that should be false for all configs

File: tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    kord_tm: -9
+    kord_tr: 9
+    kord_wz: 9
+    layout:
+    - 1
+    - 1
+    make_nh: true
+    mountain: false
+    n_split: 8
+    n_sponge: 20
```
Comment (rheacangeo) [inline tests/serialized_test_data_generation/configs/c96_6ranks_baroclinic.yml:149]: how did you decide on these?

**Review by rheacangeo — COMMENTED** [review]
Looks good! Some open questions, mostly the ones Oli brought up and the settings changed compared to the c12 case.

**Comment by rheacangeo** [comment]
the 54 ranks crashes in jenkins for an unknown reason (https://jenkins.ginko.ch/job/fv3gfs-fortran-generate-test-data/204/console) .. we might want to not include that in this round unless we can determine the problem.

**Review by ofuhrer — APPROVED** [review]

**Review by rheacangeo — APPROVED** [review]
Yay! Looks good. thanks for your patience with all the janky snafus

**Pull request merged**